src/ResourceManager/DeploymentManager/Commands.DeploymentManager/Commands/RolloutCmdletBase.cs

```diff
@@ -0,0 +1,319 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.DeploymentManager.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+
+    using Microsoft.Azure.Commands.DeploymentManager.Models;
+    using Microsoft.Azure.Commands.DeploymentManager.Utilities;
+    using Microsoft.Azure.Management.DeploymentManager.Models;
+
+    public abstract class RolloutCmdletBase : DeploymentManagerBaseCmdlet
+    {
+        private const string FailedStatus = "Failed";
+
+        private const int NoIndent = 0;
+
+        private const int ServiceUnitIndentFactor = 1;
+
+        private const int StepIndentFactor = 2;
+
+        private const int StepPropertiesIndentFactor = 3;
+
+        private const int ResourceOperationsIndentFactor = 4;
+
+        protected static void PrintVerbose(string message)
+        {
+            Console.WriteLine(message);
```

Comment:
@deveshguha you never want to use `Console.WriteLine(...)` in PowerShell -- PowerShell has separate streams that you can write to and handles their display differently.

For verbose text, you'll want to use the verbose stream, which you can write to with the method `WriteVerbose(...)`